pyproject.toml

```diff
@@ -30,13 +30,23 @@ msal = "^1"
 protobuf = ">=4"
 pip = ">=20.0.0"  # make optional once poetry doesn't auto-remove it on "simple install"
 typing_extensions = ">= 4"
-numpy = { version = "^1.20", optional = true }
+numpy = [
+    {version = ">=1.20, <1.25", python = "~3.8", optional = true},
+    {version = "^1.25", python = ">=3.9, <3.12", optional = true},
+    {version = "^1.26", python = "^3.12", optional = true},
+]
 sympy = { version = "*", optional = true }
-pandas = { version = ">=1.4", optional = true }
+pandas = [
+    { version = ">=1.4, <2.1", python = "~3.8", optional = true },
+    { version = ">=2.1", python = ">=3.9", optional = true },
+]
 backports-zoneinfo = { version = ">=0.2.1",  python = "<3.9", extras = ["tzdata"], optional = true}
 # Windows does not have a ANSI database and need tzdata
 tzdata = {version = ">=2023.3", markers = "sys_platform == 'win32'", optional = true }
-geopandas = { version = ">=0.10.0", optional = true }
+geopandas = [
+    { version = ">=0.10, <0.14", python = "~3.8", optional = true },
+    { version = ">=0.14", python = ">=3.9", optional = true },
+]
```

Comment:
Is this discovered issues?

Reply:
Oh yes, it was a pain in the b*** due to how 3.8 support typically has been dropped before 3.12 support was added, so when installing using the lock file, 💩 goes sideways... Anyway, fixed now!